src/main/java/com/teamacronymcoders/contenttweaker/modules/vanilla/resources/ItemBracketHandler.java

```diff
@@ -33,17 +33,34 @@ public IZenSymbol resolve(IEnvironmentGlobal environment, List<Token> tokens) {
 
         if (tokens.size() > 2) {
             if ("item".equalsIgnoreCase(tokens.get(0).getValue())) {
-                String blockName;
-                int meta = 0;
-                if (tokens.size() >= 5) {
-                    blockName = tokens.get(2).getValue() + ":" + tokens.get(4).getValue();
-                    if (tokens.size() >= 7) {
-                        meta = Integer.parseInt(tokens.get(6).getValue());
+                int stage = 0;
+                StringBuilder blockName = new StringBuilder();
+                StringBuilder metaValue = new StringBuilder();
+
+                for(Token tok: tokens.subList(2, tokens.size())) {
+                    if(stage == 0) { // mod-id construction
+                        if(":".equalsIgnoreCase(tok.getValue())) {
+                            blockName.append(":");
+                            stage = 1;
+                        } else {
+                            blockName.append(tok.getValue());
+                        }
+                    } else if (stage == 1) { // item name construction
+                        if(":".equalsIgnoreCase(tok.getValue())) {
+                            stage = 2;
+                        } else {
+                            blockName.append(tok.getValue());
+                        }
+                    } else { // meta value gathering
+                        metaValue.append(tok.getValue());
```

Comment:
You may want to check if the token is a Numeric token, otherwise the parseInt later down the line will throw?

Reply:
After thinking about this for a while, would it not be better for a bracket with a non-numeric meta-value to throw an error as it would be presumably an invalid item reference? If it should throw a particular error that will produce more informative error messages then I'm happy to do that, but im not sure that something like `<item:minecraft:coal:13-32>` should just silently ignore the invalid characters.

Reply:
As you like 🤷🏽